service/api_task.go

```diff
@@ -317,7 +320,36 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, currentHost *host.Host)
 	// continue must be preceded by dequeueing the current task from the
 	// queue to prevent an infinite loop.
 	for taskQueue.Length() != 0 {
-		queueItem := taskQueue.FindNextTask(spec)
+		var queueItem *model.TaskQueueItem
+		var err error
+		switch currentHost.Distro.PlannerSettings.Version {
+		case evergreen.PlannerVersionTunable:
+			if taskQueueService == nil {
```

Comment:
can we make sure this is never true by putting it in the initialization of the ui server?

Reply:
done